src/combinatorics/simple_graph/triangle/basic.lean

```diff
@@ -28,21 +33,183 @@ This module defines and proves properties about triangles in simple graphs.
 * Find a better name for `far_from_triangle_free`. Added 4/26/2022. Remove this TODO if it gets old.
 -/
 
-open finset fintype nat
-open_locale classical
+namespace finset
+variables {α : Type*} [decidable_eq α]
+
+lemma insert_comm (a b : α) (s : finset α) : insert a (insert b s) = insert b (insert a s) :=
+coe_injective $ by { push_cast, exact set.insert_comm _ _ _ }
+
+end finset
+
+namespace simple_graph
+variables {V : Type*}
+
+@[simp] lemma edge_set_top : (⊤ : simple_graph V).edge_set = {e : sym2 V | ¬ e.is_diag} :=
+by ext x; induction x using sym2.ind; simp
+
+@[simp] lemma edge_finset_top [fintype V] [decidable_eq V] [fintype (⊤ : simple_graph V).edge_set] :
+  (⊤ : simple_graph V).edge_finset = finset.univ.filter (λ e, ¬ e.is_diag) :=
+by ext x; induction x using sym2.ind; simp
+
+end simple_graph
+
+open finset fintype (card) nat
 
 namespace simple_graph
-variables {α 𝕜 : Type*} [fintype α] [linear_ordered_field 𝕜] {G H : simple_graph α} {ε δ : 𝕜}
-  {n : ℕ} {s : finset α}
+variables {α β 𝕜 : Type*} [linear_ordered_field 𝕜] {G H : simple_graph α} {ε δ : 𝕜} {n : ℕ}
+  {s : finset α}
+
+section locally_linear
+
+/-- A graph has edge-disjoint triangles if each edge belongs to at most one triangle. -/
+def edge_disjoint_triangles (G : simple_graph α) : Prop :=
+(G.clique_set 3).pairwise $ λ x y, (x ∩ y : set α).subsingleton
```

Comment:
This feels like a weird name for a proposition, as it's a noun. Perhaps `has_edge_disjoint_triangles`? Otherwise I'd expect this to return the 3-cliques that don't overlap with each other.

Reply:
cf #19201